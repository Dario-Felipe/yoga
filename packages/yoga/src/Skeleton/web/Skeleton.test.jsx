import React from 'react';
import { render } from '@testing-library/react';

import { ThemeProvider, Skeleton } from '../..';

describe('<Skeleton />', () => {
  it('should render the circular skeleton', () => {
    const { container } = render(
      <ThemeProvider>
        <Skeleton type="circular" width={64} height={64} />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render the rectangular skeleton', () => {
    const { container } = render(
      <ThemeProvider>
        <Skeleton type="rectangular" width={400} height={200} />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render the text skeleton', () => {
    const { container } = render(
      <ThemeProvider>
        <Skeleton type="text" variant="body1" width={61} />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render all text skeleton variants correctly', () => {
    const { container } = render(
      <ThemeProvider>
        <Skeleton type="text" variant="h1" width="100" />
        <Skeleton type="text" variant="h2" width="100" />
        <Skeleton type="text" variant="h3" width="100" />
        <Skeleton type="text" variant="h4" width="100" />
        <Skeleton type="text" variant="h5" width="100" />
        <Skeleton type="text" variant="body1" width="100" />
        <Skeleton type="text" variant="body2" width="100" />
        <Skeleton type="text" variant="overline" width="100" />
        <Skeleton type="text" variant="exception" width="100" />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render all background color variants correctly', () => {
    const { container } = render(
      <ThemeProvider>
        <Skeleton type="text" variant="body1" width="100" />
        <Skeleton type="text" variant="body1" width="100" color="secondary" />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
