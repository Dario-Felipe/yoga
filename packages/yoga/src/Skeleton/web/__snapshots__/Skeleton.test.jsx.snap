// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skeleton /> should render all background color variants correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100px;
  background-color: #F5F5FA;
  height: 24px;
  -webkit-animation: placeholderShimmer 1.5s ease-in-out 0.5s infinite;
  animation: placeholderShimmer 1.5s ease-in-out 0.5s infinite;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100px;
  background-color: #D7D7E0;
  height: 24px;
  -webkit-animation: placeholderShimmer 1.5s ease-in-out 0.5s infinite;
  animation: placeholderShimmer 1.5s ease-in-out 0.5s infinite;
}

<div>
  <div
    class="c0"
    color="primary"
    type="text"
    width="100"
  />
  <div
    class="c1"
    color="secondary"
    type="text"
    width="100"
  />
</div>
`;

exports[`<Skeleton /> should render all text skeleton variants correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100px;
  background-color: #F5F5FA;
  height: 56px;
  -webkit-animation: placeholderShimmer 1.5s ease-in-out 0.5s infinite;
  animation: placeholderShimmer 1.5s ease-in-out 0.5s infinite;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100px;
  background-color: #F5F5FA;
  height: 48px;
  -webkit-animation: placeholderShimmer 1.5s ease-in-out 0.5s infinite;
  animation: placeholderShimmer 1.5s ease-in-out 0.5s infinite;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100px;
  background-color: #F5F5FA;
  height: 40px;
  -webkit-animation: placeholderShimmer 1.5s ease-in-out 0.5s infinite;
  animation: placeholderShimmer 1.5s ease-in-out 0.5s infinite;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100px;
  background-color: #F5F5FA;
  height: 32px;
  -webkit-animation: placeholderShimmer 1.5s ease-in-out 0.5s infinite;
  animation: placeholderShimmer 1.5s ease-in-out 0.5s infinite;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100px;
  background-color: #F5F5FA;
  height: 28px;
  -webkit-animation: placeholderShimmer 1.5s ease-in-out 0.5s infinite;
  animation: placeholderShimmer 1.5s ease-in-out 0.5s infinite;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100px;
  background-color: #F5F5FA;
  height: 24px;
  -webkit-animation: placeholderShimmer 1.5s ease-in-out 0.5s infinite;
  animation: placeholderShimmer 1.5s ease-in-out 0.5s infinite;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100px;
  background-color: #F5F5FA;
  height: 20px;
  -webkit-animation: placeholderShimmer 1.5s ease-in-out 0.5s infinite;
  animation: placeholderShimmer 1.5s ease-in-out 0.5s infinite;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100px;
  background-color: #F5F5FA;
  height: 16px;
  -webkit-animation: placeholderShimmer 1.5s ease-in-out 0.5s infinite;
  animation: placeholderShimmer 1.5s ease-in-out 0.5s infinite;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100px;
  background-color: #F5F5FA;
  height: 12px;
  -webkit-animation: placeholderShimmer 1.5s ease-in-out 0.5s infinite;
  animation: placeholderShimmer 1.5s ease-in-out 0.5s infinite;
}

<div>
  <div
    class="c0"
    color="primary"
    type="text"
    width="100"
  />
  <div
    class="c1"
    color="primary"
    type="text"
    width="100"
  />
  <div
    class="c2"
    color="primary"
    type="text"
    width="100"
  />
  <div
    class="c3"
    color="primary"
    type="text"
    width="100"
  />
  <div
    class="c4"
    color="primary"
    type="text"
    width="100"
  />
  <div
    class="c5"
    color="primary"
    type="text"
    width="100"
  />
  <div
    class="c6"
    color="primary"
    type="text"
    width="100"
  />
  <div
    class="c7"
    color="primary"
    type="text"
    width="100"
  />
  <div
    class="c8"
    color="primary"
    type="text"
    width="100"
  />
</div>
`;

exports[`<Skeleton /> should render the circular skeleton 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 64px;
  height: 64px;
  background-color: #F5F5FA;
  border-radius: 9999px;
  -webkit-animation: placeholderShimmer 1.5s ease-in-out 0.5s infinite;
  animation: placeholderShimmer 1.5s ease-in-out 0.5s infinite;
}

<div>
  <div
    class="c0"
    color="primary"
    height="64"
    type="circular"
    width="64"
  />
</div>
`;

exports[`<Skeleton /> should render the rectangular skeleton 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 400px;
  height: 200px;
  background-color: #F5F5FA;
  -webkit-animation: placeholderShimmer 1.5s ease-in-out 0.5s infinite;
  animation: placeholderShimmer 1.5s ease-in-out 0.5s infinite;
}

<div>
  <div
    class="c0"
    color="primary"
    height="200"
    type="rectangular"
    width="400"
  />
</div>
`;

exports[`<Skeleton /> should render the text skeleton 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 61px;
  background-color: #F5F5FA;
  height: 24px;
  -webkit-animation: placeholderShimmer 1.5s ease-in-out 0.5s infinite;
  animation: placeholderShimmer 1.5s ease-in-out 0.5s infinite;
}

<div>
  <div
    class="c0"
    color="primary"
    type="text"
    width="61"
  />
</div>
`;
